Wikipedia/Code/UINavigationBar+WMFImageGalleryGradient.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+import QuartzCore
+
+extension UINavigationBar {
+    
+    func wmf_applyGalleryTopGradientBackground() {
+        if (!CGRectIsNull(bounds)){
+            if (CGRectGetWidth(bounds) > 0 && CGRectGetHeight(bounds) > 0){
+                setBackgroundImage(wmf_galleryTopGradientBackgroundImage(), forBarMetrics: .Default)
+            }
+        }
+    }
+    
+    func wmf_galleryTopGradientBackgroundImage() -> UIImage {
+        UIGraphicsBeginImageContextWithOptions(bounds.size, false, UIScreen.mainScreen().scale)
```

Comment:
you could just draw this as 1px wide and stretch the image. if performance isn't an issue, probably not worth the change.